controllers/dspipeline_controller.go

```diff
@@ -397,7 +409,16 @@ func (r *DSPAReconciler) handleReadyCondition(
 
 }
 
-func (r *DSPAReconciler) GenerateStatus(ctx context.Context, dspa *dspav1alpha1.DataSciencePipelinesApplication) ([]metav1.Condition, error) {
+func (r *DSPAReconciler) GenerateStatus(ctx context.Context, dspa *dspav1alpha1.DataSciencePipelinesApplication, params *DSPAParams) ([]metav1.Condition, error) {
+	databaseAvailable := r.buildCondition(config.DatabaseAvailable, dspa, config.DatabaseAvailable)
+	if r.isDatabaseAccessible(ctx, dspa, params) {
+		databaseAvailable.Status = metav1.ConditionTrue
+	}
+
+	objStoreAvailable := r.buildCondition(config.ObjectStoreAvailable, dspa, config.ObjectStoreAvailable)
+	if r.isObjectStorageAccessible(ctx, dspa, params) {
+		objStoreAvailable.Status = metav1.ConditionTrue
+	}
```

Comment:
It looks like we're doing this call twice, can we just pass the values we obtained [here](https://github.com/gmfrasca/data-science-pipelines-operator/blob/6992e9b4678a973e529d5a10c5382645f20089d8/controllers/dspipeline_controller.go#L224)?

Reply:
sure, but was trying to avoid it as it pollutes the function signatures.  That said, these are 'expensive' calls so probably better to pass them in in this case

Reply:
updated